### PR TITLE
Replace argparse with Typer library 

### LINKED ISSUE #7: Add basic tests
Currently we don't really have any tests that verify our command-line options work as expected. Up until now we've just been adding different CLI arguments to the `launch.json`. I think that we should restructure the `/tests` folder to actually have some tests in it, instead of just a bunch of [doxy, config and header] files. This would be helpful moving forward, such that when we add new features we can easily run some tests to know we didn't break other features.

### Objectives
- Research/use `pytest` (or similar) to instrument some test cases
- Create a test case for each *useful* command-line argument, feature or use-case, for instance: `--dont-reset`, `part`, `--dry-run`, mismatched version numbers, etc.
  - This will mean that we will have multiple/many configuration files and source files for the tests in order to support the aforementioned test cases.

### LINKED ISSUE #8: Add some kind-of 'display' command line argument
Perhaps add something like: `--display-versions` that would go into the configuration and get the current version of all components and all files. Then print the versions out in a table/list.

### PR DESCRIPTION
**Pull request by travisrecupero**
# Why not argparse?
Say we had two arguments `--bump` and `--display`. We can say that they both require a `--config-folder` and `--bump` requires an extra argument `--part`. If `--part` is _required_, then we need to create a subparser where `--display` doesn't need a `--part`. This leads to a messy hierarchy of optional and/or required arguments.

# Typer
- Typer is a library for creating CLI applications in python. 
- Documentation for Typer can be found here https://typer.tiangolo.com/.
- Our hierarchal issue gets solved by the use of _decorators_. A _decorator_ is a function that takes another function and extends the behavior of the latter function without explicitly modifying it. For example:
```python
@app.command()
def display_version(config: str):
    """
    Print out all components respective versions
    """
    if os.path.exists(config):
        print_versions(config, versions)
```
Here we extend the behavior of the `command` function, which will create the `display_version` command for us. 

### An easy way to think about this is: 
        Function name is a command
        Function parameter is an argument

# Changes
- replaced argparse with Typer
- added simple test cases

Fixes #8 
Fixes #7

### CODE REVIEW

File: sample_files/Doxyfile [hunk 1]
```diff
@@ -39,7 +39,7 @@ PROJECT_NAME           = rascal_lib
 # control system is used.
 
 
-PROJECT_NUMBER =13.1.1
+PROJECT_NUMBER = 15.4.0
```
Comment (coppolat1) [inline sample_files/Doxyfile]: We probably shouldn't commit any changes to the files in /sample_files

File: sample_files/sample-input-file.h [hunk 2]
```diff
@@ -15,7 +15,7 @@
  * Major version MUST be incremented if any backwards incompatible changes are introduced to the public API. It MAY also include
  * minor and patch level changes. Patch and minor version MUST be reset to 0 when major version is incremented.
  */
-#define LIBNAME_VERSION_MAJOR        (10U)
+#define LIBNAME_VERSION_MAJOR        (15U)
```
Comment (coppolat1) [inline sample_files/sample-input-file.h]: We probably shouldn't commit any changes to the files in /sample_files

File: bumpCversion.py [hunk 3]
```diff
@@ -94,127 +131,90 @@ def get_config(config_file):
...
+                versions.add(str(filetype.version_number))
+                component.paths.pop(0)
+            if len(versions) == 1:
+                versions.clear()
+                continue
+            else:
+                print("Multiple Verisons Found -> " + str(versions))
+                raise(VersionException(Exception))
+
+    return 1
```
Comment (coppolat1) [inline bumpCversion.py]: Should this be `return True` instead of `return 1`?

**Review by coppolat1 — COMMENTED** [review]
Looks good! Just a couple small considerations.

File: sample_files/sample-input-file.h (hunk 2 above)
Comment (travisrecupero) [reply sample_files/sample-input-file.h]: I will make a commit to change all the sample files to have the same version number, then ill test/add the dir to the .gitignore. It seems the .gitignore isnt actually ignoring the sample files.

File: bumpCversion.py (hunk 3 above)
Comment (travisrecupero) [reply bumpCversion.py]: Changed

**Pull request merged**